library/src/main/java/com/datatorrent/lib/io/block/ReaderContext.java

```diff
@@ -190,11 +190,11 @@ protected Entity readEntity() throws IOException
           if (bytesRead == -1) {
             break;
           }
-          strBuffer = new String(buffer);
+          bufferStr = new String(buffer,0, bytesRead);
         }
 
-        while (posInStr < strBuffer.length()) {
-          char c = strBuffer.charAt(posInStr);
+        while (posInStr < bufferStr.length()) {
+          char c = bufferStr.charAt(posInStr);
           if (c != '\r' && c != '\n') {
             tmpBuilder.write(c);
```

Comment:
This is lossy write to tmpBuilder using ByteArrayOutputStream.write(int). byte[] to String conversion (line 193) is based on default encoding. In case of UTF-16 (or UTF-32) as default encoding encoding, this will lead to lose of data...

After looking at the code little more, I found that we can get rid of tmpBuilder and emptyBuilder